src/Data-MessagePack-Stream.xs

```diff
@@ -38,11 +38,14 @@ static SV* decode_msgpack_object(msgpack_object* obj) {
         case MSGPACK_OBJECT_NEGATIVE_INTEGER:
             res = newSViv(obj->via.i64);
             break;
-        case MSGPACK_OBJECT_DOUBLE:
-            res = newSVnv(obj->via.dec);
+        case MSGPACK_OBJECT_FLOAT:
+            res = newSVnv(obj->via.f64);
             break;
-        case MSGPACK_OBJECT_RAW:
-            res = newSVpvn(obj->via.raw.ptr, obj->via.raw.size);
+        case MSGPACK_OBJECT_BIN:
+            res = newSVpvn(obj->via.bin.ptr, obj->via.bin.size);
+            break;
+        case MSGPACK_OBJECT_STR:
+            res = newSVpvn(obj->via.str.ptr, obj->via.str.size);
```

Comment:
I'm very much a novice when it comes to XS, but why isn't `newSVpvn_utf8` used here?  Since the STR type is utf8, shouldn't the `SVf_UTF8` flag be set?